src/app/layout.tsx

```
@@ -26,16 +29,41 @@ const notoSans = Noto_Sans({
-    <html lang="en" className={`${spaceGrotesk.variable} ${notoSans.variable}`} suppressHydrationWarning>
+    <html
+      lang="en"
+      className={`${spaceGrotesk.variable} ${notoSans.variable} ${initialThemeClass}`}
+      suppressHydrationWarning>
       <head>
         <Script id="init-theme" strategy="beforeInteractive">
-          {`(function(){try{var m=document.cookie.match(/(?:^|; )theme=([^;]+)/);var v=m?decodeURIComponent(m[1]):null; if(v&&v!=='system'){document.documentElement.classList.add(v);} else if(window.matchMedia&&window.matchMedia('(prefers-color-scheme: dark)').matches){document.documentElement.classList.add('dark');}}catch(e){}})();`}
+          {buildInitThemeScript()}
         </Script>
+        <script
+          dangerouslySetInnerHTML={{
```

Biome (2.1.2)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)